plugin/diagnostics.py

```diff
@@ -96,7 +97,8 @@ def on_selection_modified_async(self) -> None:
 
     def show_diagnostics_status(self, diagnostic: Diagnostic) -> None:
         self.has_status = True
-        self.view.set_status('lsp_diagnostics', diagnostic.message)
+        spaced_message = re.sub(r'(\S\n)(\S)', r'\1 \2', diagnostic.message)
```

Comment:
Wouldn't it make sense to just remove the newline: `r'(\S)\n(\S)'`?
I guess it wouldn't make a difference in practice but I feel it's "more correct".

Reply:
`diagnostic.message.replace("\n", " ")` would this be the equivalent of the regex you typed?

Reply:
Not if you want this behavior:
> This PR inserts a space into the string passed to set_status where a line in a diagnostic message ends without trailing whitespace and the next line starts without leading whitespace.

Reply:
I would keep it simple and suggest using a regular replace,
unless @frou insist on using a regex solution.

In that case , just put a comment above the regex

Reply:
If the message already has a space before and/or after the newline, doing a simple string replace would result in there being >1 spaces between the words.

Reply:
I feel like this extra text after `\n` pollutes the status bar too much anyway. Perhaps strip away everything after the `\n`?

Reply:
Why make an assumption about how servers arrange the useful content in their message? It no doubt varies widely. Even in my example, the latter part is definitely helpful.

Reply:
OK, makes sense.